resources/pages/SearchSong.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import { apiUrl } from "../js/App";
+import { useParams } from "react-router-dom";
+import SearchBar from "../components/SearchBar";
+import { Grid, Stack } from "@mui/material";
+import CircularProgress from "@mui/material/CircularProgress";
+import SongSearchCard from "../components/SongSearchCard";
+import { useDebounce } from "@uidotdev/usehooks";
+
+function Search() {
+    const [searchTerm, setSearchTerm] = useState("");
+    const [isLoading, setIsLoading] = useState(false);
+    const [searchResults, setSearchResults] = useState([]);
+    const debouncedSearchTerm = useDebounce(searchTerm, 500);
+    const { id } = useParams();
+
+    // const handleSearch = async () => {
+    //     setIsLoading(true);
+    //     const response = await axios.get(`${apiUrl}/${id}/songs/search`, {
+    //         params: {
+    //             q: searchTerm,
+    //         },
+    //     });
+    //     const data = response.data;
+    //     setSearchResults(data);
+    //     setIsLoading(false);
+    // };
+
+    // useEffect(() => {
+    //     setSearchTerm("");
+    // }, [searchResults]);
+
+    const handleInput = (e) => {
+        setSearchResults([]);
+        setSearchTerm(e.target.value);
+    };
+
+    useEffect(() => {
+        const handleSearch = async () => {
+            setIsLoading(true);
+            if (debouncedSearchTerm) {
+                const response = await axios.get(
+                    `${apiUrl}/${id}/songs/search`,
+                    {
+                        params: {
+                            q: searchTerm,
+                        },
+                    }
+                );
+                const data = response.data;
+                setSearchResults(data);
+            }
+            setIsLoading(false);
+        };
+        handleSearch();
+    }, [debouncedSearchTerm]);
```

Comment:
Tu pourrais faire un hook "useSearchSong" afin de réduire le nombre de lignes dans ton composant

Reply:
Created folder `hooks `where I put `useSearchSong.js`. Quantity of lines in `SearchSong.jsx` decreased dramatically.